core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1920,11 +1920,7 @@ private DefaultServerConfig buildServerConfig(List<ServerPort> serverPorts) {
                 ChannelUtil.applyDefaultChannelOptions(
                         childChannelOptions, idleTimeoutMillis, pingIntervalMillis);
 
-        ServerErrorHandler errorHandler = this.errorHandler;
-        if (errorHandler != ServerErrorHandler.ofDefault()) {
-            // Ensure that ServerErrorHandler never returns null by falling back to the default.
-            errorHandler = errorHandler.orElse(ServerErrorHandler.ofDefault());
-        }
```

Comment:
I agree, but I think we need to do this in `VirtualHostBuilder` though, because every `VirtualHost` now has a `ServerErrorHandler`. We could update `VirtualHostBuilder.errorHandler()` instead.

Reply:
It's handled here:
https://github.com/line/armeria/pull/4716/files#diff-cd2df9e0b65586c6cbd6c95b5cd88d4460b02e78389a5011192119104160956eR1162-R1165

Reply:
I realized that it isn't handled correctly before the @trustin's comment is addressed:
https://github.com/line/armeria/pull/4716/files#r1127644667